src/write_gcm.c

```diff
@@ -3105,6 +3146,53 @@ static void wg_json_Metric(json_ctx_t *jc,
   wg_json_map_close(jc);
 }
 
+static int wg_metric_value_type_from_wg_payload_t(
+    const wg_payload_t *element) {
+  int metric_value_type = -1;
+  for (int i = 0; i < element->key.num_metadata_entries; ++i) {
+    wg_metadata_entry_t *entry = &element->key.metadata_entries[i];
+    if (strcmp(entry->key, custom_metric_value_type) == 0) {
+      if (strcmp(entry->value.value_text, "INT64") == 0) {
+        metric_value_type = METRIC_VALUE_TYPE_INT64;
+      } else if (strcmp(entry->value.value_text, "DOUBLE") == 0) {
+        metric_value_type = METRIC_VALUE_TYPE_DOUBLE;
+      } else if (strcmp(entry->value.value_text, "BOOL") == 0) {
+        metric_value_type = METRIC_VALUE_TYPE_BOOL;
+      } else {
+        WARNING("write_gcm: unsupported metric value type provided: %s", 
+                entry->value.value_text);
+      }
+      break;
+    }
+  }
+
+  // Failover to sane defaults for each DS type.
+  if(metric_value_type == -1) {
+    switch (element->values[0].ds_type) {
+      case DS_TYPE_GAUGE:
+        metric_value_type = METRIC_VALUE_TYPE_DOUBLE;
+        break;
+      case DS_TYPE_DERIVE:
+      case DS_TYPE_COUNTER:
+        metric_value_type = METRIC_VALUE_TYPE_INT64;
+        break;
+    }
+  }
+
+  return metric_value_type;
+}
+
+static char *wg_metric_value_type_to_string(int metric_value_type) {
+  if (metric_value_type < 0 || 
+      metric_value_type > STATIC_ARRAY_SIZE(metric_value_type_strings)) {
+    WARNING("write_gcm: invalid metric value type converted to a string: %d", 
```

Comment:
Maybe even `ERROR` (since this would be a bug in the code).

Reply:
Done.